gluonnlp/criterion/language_model.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=
+"""Language model criterion."""
+
+__all__ = ['ActivationRegularizationLoss', 'TemporalActivationRegularizationLoss']
+
+from mxnet import nd
+
+class ActivationRegularizationLoss(object):
+    r"""Computes Activation Regularization Loss. (alias: AR)
+
+    The formulation is as below:
+
+    .. math::
+
+        L = \alpha L_2(h_t)
+
+    where $L_2(\dot) = {||\dot||}_2$, $h_t$ is the output of the RNN at timestep t.
+    $\alpha$ is scaling coefficient.
+
+    The implementation follows the work:
+
+    @article{merity2017revisiting,
+      title={Revisiting Activation Regularization for Language RNNs},
+      author={Merity, Stephen and McCann, Bryan and Socher, Richard},
+      journal={arXiv preprint arXiv:1708.01009},
+      year={2017}
+    }
+
+    Parameters
+    ----------
+    alpha : float, default 0
+        The scaling coefficient of the regularization.
+
+    Inputs:
+        - **states**: the stack outputs from RNN,
+        which consists of output from each time step (TNC).
+
+    Outputs:
+        - **loss**: loss tensor with shape (batch_size,). Dimenions other than
+          batch_axis are averaged out.
+    """
+    def __init__(self, alpha=0):
+        self._alpha = alpha
+
+    def __repr__(self):
+        s = 'ActivationRegularizationLoss {name} (alpha={alpha})'
+        return s.format(name=self.__class__, alpha=self._alpha)
```

Comment:
This would print as `ActivationRegularizationLoss ActivationRegularizationLoss  (alpha=xx)`